web/components/feed/feed-context.tsx

```diff
@@ -0,0 +1,18 @@
+import { createContext, ReactNode, useContext } from 'react'
+import { CommentTipMap, useTipTxns } from 'web/hooks/use-comment-tips'
+
+const FeedContext = createContext<CommentTipMap | null | undefined>(null)
```

Comment:
I think your implementation using Context is good. Clean and reasonable.

However, we don't use context anywhere else in our codebase. I'm concerned about introducing a new pattern because it adds complexity.

Secondly, I'm not a fan of context in general. It's an implicit parameter and those are bad. Someone who tries to use the FeedComment elsewhere will be mysteriously missing tip data.

I think part of the problem here is that the feed items code is complex and not following normal react structure: we compute a list of activity items as a model first and then pass them dynamically into the relevant feed item components. This makes it hard to just pass props all the way through.

However, I think passing the tips data all the way from `ContractActivity` is still a better solution. Just follow how `comments` are passed through. It's probably less of a code change than setting up this context.

One alternate solution would be to let each comment fetch its own data, which is then cached with react query. It would have to query the txns for its own comment id. I think this would be an ok solution as well.

Something like:
```
export const useCommentTips = (commentId: string) => {
  const result = useDatabaseSnapshot(
    ['comment-tips', commentId],
    // insert query for 'TIP' txns involving commentId,
    { subscribe: true, includeMetadataChanges: true }
  )
  return result.isLoading ? undefined : result.data
}
```

Any thoughts?

Reply:
Using react-query is interesting. But if we let each comment fetch it's own data, we may as well just have add a new firebase listener to each comment directly instead of using react-query. Each query would be unique anyways, so we wouldn't be deduping, and we aren't using the queries loading/error states. It would make more sense to use react-query if we want to grab all the tips in the contract and just filter it on a per-comment level.

We also currently don't use react-query much and it runs into similar code complexity issues as context (like, devs need to learn about query keys).

Reply:
> passing the tips data all the way is probably less of a code change than setting up this context

Depends how you count. The prop gets added in 20+ different places

Reply:
Good point on just using a firebase listener for the comment tip. That could be the best solution. If you want to be able to sort over all the tips, you could just fetch all the txns again from the parent component.